### PR TITLE
refactor: use a controller model for wizard flows

### PR DESCRIPTION
**Pull request by rockstar**
This patch uses a Controller pattern for managing the flow of creating a
new bucket. The value is that the TreeView nodes aren't storing
unrelated logic for other extension functionality. This pattern, applied
across all of these flows, should organize the code in a more
maintainable way.

### CODE REVIEW

File: src/controllers/AddBucketController.ts [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    constructor(
+        private instance: IInstance,
+        private context: vscode.ExtensionContext) {
+        console.log("AddBucketController constructor")
+        const addBucketView = new AddBucketView(this.context)
+        addBucketView.show(this.addBucketCallback.bind(this))
+    }
+
+    private async addBucketCallback(name : string, duration : number | undefined) : Promise<void> {
+        // XXX: rockstar (13 Sep 2021) - This makes me irrationally annoyed. The
```
Comment (jsternberg) [inline src/controllers/AddBucketController.ts:78]: Can we make an issue about this? I don't know if this is by design or just unfortunately neglected.

File: src/controllers/AddBucketController.ts [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+}
+
+/* Control the flow for the Add/Edit bucket flow, including presenting views, storing
+ * process state, cleaning up resources, and restoring state.
+ */
+export class AddBucketController {
+    constructor(
+        private instance: IInstance,
+        private context: vscode.ExtensionContext) {
+        console.log("AddBucketController constructor")
```
Comment (jsternberg) [inline src/controllers/AddBucketController.ts:72]: Are these `console.log()` statements intended?

File: src/controllers/AddBucketController.ts [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+        await bucketsApi.postBuckets({
+            body: {
+                orgID,
+                name,
+                retentionRules
+            }
+        })
+        vscode.commands.executeCommand('influxdb.refresh')
+    }
+}
```
Comment (jsternberg) [inline src/controllers/AddBucketController.ts:105]: Ensure that files end in a newline. If your editor doesn't do this by default, you can usually configure it to.

Comment (rockstar) [reply src/controllers/AddBucketController.ts:105]: Linting should have caught this, too. However, I don't care if files end in a newline.

**Pull request merged**